core/typedef.go

```diff
@@ -72,6 +72,50 @@ func (fn *Function) WithArg(name string, typeDef *TypeDef, desc string, defaultV
 	return fn
 }
 
+func (fn *Function) IsSubtypeOf(otherFn *Function) bool {
+	if fn == nil || otherFn == nil {
+		return false
+	}
+
+	// check return type
+	if !fn.ReturnType.IsSubtypeOf(otherFn.ReturnType) {
+		return false
+	}
+
+	// check args
+	for i, otherFnArg := range otherFn.Args {
+		/* TODO: with more effort could probably relax and allow:
+		* arg names to not match (only types really matter in theory)
+		* mismatches in optional (provided defaults exist, etc.)
+		* fewer args in interface fn than object fn (as long as the ones that exist match)
```

Comment:
True, that's another option. I guess it comes down to whether we want to support matches like this:
* Object - `fn(arg1 string, arg2 int)`
* Interface - `fn(arg2 int, arg1 string)`

Or matches like this:
* Object - `fn(arg1 string, arg2 int)`
* Interface - `fn(blah1 string, blah2 int)`

I *think* it would be an exclusive choice between those; I'm not immediately seeing how we could support both arbitrary name and arbitrary order.

If I was forced to choose now, I'd probably go with order mattering but not name. Can't really justify why fully, I guess it just feels more likely to choose a slightly different name for an arg but still wanting to match than it does to choose a different order of args. But atm it's just not supported so we can decide on that if/when it arises in practice.